magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/HooksService.scala

```diff
@@ -41,7 +41,9 @@ class HooksService(config: Config, webHookActor: ActorRef, authClient: AuthApiCl
       entity(as[WebHook]) { hook =>
         DB localTx { session =>
           HookPersistence.create(session, hook) match {
-            case Success(result)    => complete(result)
+            case Success(result) =>
+              webHookActor ! WebHookActor.InvalidateWebhookCache
```

Comment:
Isn't this a race condition?  The Invalidate message is sent before the transaction commits.  If the actor wakes up and receives the message before the commits, too, it won'ot see the new hook that was added.

Reply:
Yeah that makes sense... I think it's being shielded by the big webhook actor only processing one thing at a time, but it should be after.